shell/browser/ui/message_box_mac.mm

```diff
@@ -76,15 +76,16 @@
     [[ns_buttons objectAtIndex:settings.default_id] setKeyEquivalent:@"\r"];
   }
 
-  // Bind cancel id button to escape key if there is more than one button
-  if (button_count > 1 && settings.cancel_id >= 0 &&
-      settings.cancel_id < button_count) {
-    [[ns_buttons objectAtIndex:settings.cancel_id] setKeyEquivalent:@"\e"];
-  }
+  if (button_count > 1) {
+    // Bind cancel id button to escape key if there is more than one button.
+    if (settings.cancel_id >= 0 && settings.cancel_id < button_count) {
+      [[ns_buttons objectAtIndex:settings.cancel_id] setKeyEquivalent:@"\e"];
+    }
 
-  // TODO(@codebytere): This behavior violates HIG & should be deprecated.
-  if (settings.cancel_id >= 0 && settings.cancel_id == settings.default_id) {
-    [[ns_buttons objectAtIndex:settings.default_id] highlight:YES];
+    // TODO(@codebytere): This behavior violates HIG & should be deprecated.
+    if (settings.cancel_id >= 0 && settings.cancel_id == settings.default_id) {
+      [[ns_buttons objectAtIndex:settings.default_id] highlight:YES];
+    }
```

Comment:
_(nit)_ both inner `if` clauses test for `settings.cancel_id >= 0`, which can be moved to the outer `if`:

```c++
if (button_count > 1 && settings.cancel_id >= 0) {
  // Bind cancel id button to escape key if there is more than one button
  if (settings.cancel_id < button_count)
    [[ns_buttons objectAtIndex:settings.cancel_id] setKeyEquivalent:@"\e"];

  // TODO(@codebytere): This behavior violates HIG & should be deprecated
  if (settings.cancel_id == settings.default_id)
    [[ns_buttons objectAtIndex:settings.default_id] highlight:YES];
}
```